Source/Dafny/Compiler-java.cs

```diff
@@ -1978,7 +1978,9 @@ private class GenericArrayElementLvalue : ILvalue {
 
     protected void EmitToString(TargetWriter wr, Expression arg) {
       if (arg.Type.IsArrowType) {
-        wr.Write(IdName(((IdentifierExpr) ((ConcreteSyntaxExpression)arg).ResolvedExpression).Var) + " == null ? null : \"Function\"");
+        var expr = ((ConcreteSyntaxExpression)arg).ResolvedExpression;
```

Comment:
There's no reason to believe `arg` is a `ConcreteSyntaxExpression`. The proper way to do this is:

    var expr = arg.Resolved;

Reply:
Fixed. Still learning this type hierarchy.